src/format/locales.rs

```diff
@@ -31,3 +31,67 @@ pub(crate) fn d_t_fmt(locale: Locale) -> &'static str {
 pub(crate) fn t_fmt(locale: Locale) -> &'static str {
     locale_match!(locale => LC_TIME::T_FMT)
 }
+
+macro_rules! expand_ampm {
+    ($item:ident) => {
+        pure_rust_locales::$item::LC_TIME::T_FMT_AMPM
+    };
+}
+
+/// hardcoded `locale_match!` fallback for `T_FMT_AMPM`.
+///
+/// This is because `locale_match!` cannot be used when a locale misses a field.
+/// see [related issue](https://github.com/cecton/pure-rust-locales/issues/4) for details.
+macro_rules! locale_match_ampm {
+    ($locale:expr => ( $($item:ident),* $(,)? )) => {
+        match $locale {
+            $(Locale::$item => {
+                let t_fmt_ampm = expand_ampm!($item);
+
+                if t_fmt_ampm.is_empty() {
+                    expand_ampm!(en_US)
+                } else {
+                    t_fmt_ampm
+                }
+            },)*
+
+            // default fallback is en_US
+            _ => expand_ampm!(en_US),
+        }
+    }
+}
+
+pub(crate) fn t_fmt_ampm(locale: Locale) -> &'static str {
```

Comment:
I think this should get a clear comment to explain in high-level terms what it's trying to achieve, as that is pretty unclear to me. I'm not sure I agree that the stacked macro approach is the best option here...

Reply:
alright, i think description for it would be similar to `locale_match_ampm!` macro, but will try splitting it up for high and low level terms.

Reply:
added more docstring at 3f0aafdaac76cdfa3bbbbcf623749b55c4af5f94. could you take a look?